configure.ac

```diff
@@ -10,12 +14,225 @@ AM_INIT_AUTOMAKE(subdir-objects)
 AM_MAINTAINER_MODE([disable])
 AX_AM_MACROS
 
+# Update this value for every release: (A:B:C will map to foo.so.(A-C).C.B)
+# http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+PVECLIB_SO_VERSION=1:4:0
+AC_SUBST(PVECLIB_SO_VERSION)
+
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
+# This directive is to avoid buggy libtool that don't add the -Wl,--no-as-needed
+# directive in the correct position of LDFLAGS
+LDFLAGS="$LDFLAGS -Wl,--no-as-needed -lc"
+
 # Checks for header files.
 AC_CHECK_HEADERS([altivec.h stddef.h stdint.h unistd.h stdio.h fenv.h float.h math.h])
 
+#############################################################################
+
+CNAME=`basename "$CC"`
+AM_CPPFLAGS="-DNDEBUG"
+AM_CFLAGS="-m64 -O3"
+
+## PowerPC-64 Specific options
+## Save CFLAGS
+SAVED_CFLAGS="$CFLAGS"
+
+ALTIVEC_CFLAGS="-mcpu=970 -maltivec"
+POWER_F128_CFLAGS="-mfloat128"
+POWER6_CFLAGS="-mcpu=power6 -maltivec"
+POWER7_CFLAGS="-mcpu=power7"
+POWER8_CFLAGS="-mcpu=power8"
+POWER9_CFLAGS="-mcpu=power9 -mfloat128"
+
+##### Compiler default includes ALTIVEC #####
+AC_SUBST([PVECLIB_DEFAULT_CFLAG], [""])
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports ALTIVEC])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_VMX.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_DEFAULT_CFLAG], [""]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_ALTIVEC], [1], [Disable ALTIVEC])
+)
+
+if test "$retval" != "0"; then
+	AC_SUBST([PVECLIB_DEFAULT_CFLAG],[POWER8_CFLAGS])
+fi
+
+##### Compiler default includes DFP #####
+
+CFLAGS=$AM_CFLAGS
+AC_MSG_CHECKING([if $CNAME default supports DFP])
+AC_LINK_IFELSE(
+	[AC_LANG_SOURCE([`cat testprograms/test_ppc_DFP.c`])],
+	[AC_MSG_RESULT([yes])]
+		AC_SUBST([PVECLIB_DEFAULT_CFLAGS], [""]),
+	[AC_MSG_RESULT([no])]
+		AC_DEFINE([PVECLIB_DISABLE_DFP], [1], [Disable DFP])
+)
+
+if test "$retval" != "0"; then
+	AC_SUBST([PVECLIB_DEFAULT_CFLAGS],[POWER8_CFLAGS])
```

Comment:
Why does DFP get us to POWER8?

Reply:
Because POWER7 will fail on a LE system and we don't know that yet.
For BE system the damn compilers default to POWER4 which is completely useless.
It is a complete mess and we are just trying to get to a base target that works.

I am not ready to deal with 970 and POWER6 and Jeff Scheel keeps telling me to forget BE altogether. So this will stay and maybe look again later.